### DISCUSSION TITLE
Renovate only patches package-lock.json, not package.json

### DISCUSSION THREAD

**Discussion by jessehouwing** [renovatebot/renovate#20855] [Request Help] [answered]
### How are you running Renovate?

Self-hosted on GitHub Actions on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.0.0

### If you're self-hosting Renovate, select which platform you are using.

Windows

### If you're self-hosting Renovate, tell us what version of the platform you run.

Github.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

see:
https://github.com/microsoft/azure-devops-extension-tasks/pull/381
https://github.com/microsoft/azure-devops-extension-tasks/actions/runs/4384446516/jobs/7675931768

As of 35.0.0 renovate only patches the `package-lock.json`, but not the `package.json`.

Last successful run that created pull requests did patch both the package and lock files:
https://github.com/microsoft/azure-devops-extension-tasks/actions/runs/4256868933/jobs/7406283477

### Relevant debug logs

<details><summary>Logs</summary>

```

```

</details>

[log.txt](https://github.com/renovatebot/renovate/files/10942198/log.txt)

### Have you created a minimal reproduction repository?

I have read the minimal reproductions documentation and linked to such a repository in the bug description

**Comment by rarkins** [comment]
Which repo is the minimal reproduction?

**Comment by jessehouwing** [comment]
it's not minimal-minimal yet. Haven't been able to create one, but it happens here:
https://github.com/microsoft/azure-devops-extension-tasks

Pinned fork at the impacted version:
https://github.com/jessehouwing/azure-devops-extension-tasks-backup

Specifically, when trying to patch this project (which is pretty minimal by itself):
https://github.com/microsoft/azure-devops-extension-tasks/tree/main/BuildTasks/TfxInstaller/v4

And used to work prior to 35.0.0.

The call from the root of the repo is:

```
npm install renovate@^35
npm exec -- renovate --platform=github --include-forks=true --enabled-managers="npm,azure-pipelines,github-actions" Microsoft/azure-devops-extension-tasks
```

Reverting to 34 doesn't generate just a lock-file branch, but actually proceeds to patch (some) of the package.json files:

```
npm install renovate@^34
npm exec -- renovate --platform=github --include-forks=true --enabled-managers="npm,azure-pipelines,github-actions" Microsoft/azure-devops-extension-tasks
```

That is the only diff between working and non-working.

**Comment by jessehouwing** [comment]
I'm proceeding in manually patching the repo, as I currently don't have the time to undress this repo to seek a more minimal reproduction.

**Comment by jessehouwing** [comment]
Here is a fork that's at the impacted version:
https://github.com/jessehouwing/azure-devops-extension-tasks-backup

**Comment by rarkins** [comment]
It's a pretty heavy repro (20+ lockfiles needing separate update in the same PR) so hard to debug

**Comment by rarkins** [comment]
Is this the _wrong_ PR reproduced? https://github.com/renovate-reproductions/20846/pull/1/files

The one you linked to above now has zero files and was autoclosed

**Comment by rarkins** [comment, marked as answer]
I think I know the answer to this.

- You're running with v35
- v35 included a change where it now defaults to rangeStrategy=update-lockfile when rangeStrategy=auto, and you are using `config:base` so that's what you got
- As a result, for the first time ever, you are getting "lock file only" PRs when you have ranged dependencies in `package.json`

It's not a mistake or a bug. If you want to avoid lockfile-only updates, configure `rangeStrategy=replace` to revert back to old behavior.

**Reply by rarkins** [reply]
Converted to a Discussion and marking this as the answer

**Reply by jessehouwing** [reply]
I'll probably w;not be alone :).

**Reply by jessehouwing** [reply]
```
    {
      "rangeStrategy": "replace",
       "matchDatasources": ["npm"]
    },
```

Seems to have made things 'normal' again. Why would I use the other mode over this one?

**Reply by rarkins** [reply]
You would use the new default if you want your direct dependencies to be up to date. With a lock file, they are "held back" to whatever they were last locked to.

**Reply by jessehouwing** [reply]
But in this case the updates *were* for direct dependencies...

I can see how the lock file would update indirect dependencies in this case...

I'm also seeing a few updates not coming through at the moment. Haven't looked at the exact logfiles yet, but there are a few I would expect updates for.

**Reply by rarkins** [reply]
I also meant they're direct dependencies. And naturally when direct dependencies are updated in a lock file there can be necessary indirect dependency updates too - that's up to npm.

**Reply by jessehouwing** [reply]
I set it to "widen" now, and now I'm getting the updates I was expecting.

**Reply by rarkins** [reply]
I would advise against doing that in most cases.

Anyway, this topic is answered and done, let's not confuse it by changing topic.

**Reply by FabianFrank** [reply]
@rarkins I'm seeing what seems like a related issue with the `auto` behavior. renovate updates not only the resolved version, but also the requested range in package-lock.json:
<img width="1304" alt="Screenshot 2023-07-15 at 11 10 12 AM" src="https://github.com/renovatebot/renovate/assets/691152/d862fe43-44bb-46bd-86f4-47da42095bcd">

However, a subsequent `npm install` will set the requested version in `package-lock.json` back to the range that's in `package.json` resulting in the following diff:
```
diff --git a/package-lock.json b/package-lock.json
index 8db29b2b..b1fe4a9e 100644
--- a/package-lock.json
+++ b/package-lock.json
@@ -13,7 +13,7 @@
       ],
       "dependencies": {
         "ajv": "^8.12.0",
-        "patch-package": "^7.0.2"
+        "patch-package": "^7.0.0"
       },
       "devDependencies": {
         "@types/eslint": "^8.37.0",
```

IMHO the correct behavior would be to not update the requested range in `package-lock.json` if `package.json` is not being updated. I'm using the recommended config:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:recommended"]
}
```
